packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -52,17 +61,41 @@ export interface TreeNodeApi {
 		value: unknown,
 		schema: TSchema,
 	): value is NodeFromSchema<TSchema>;
+
+	/**
+	 * Gets the child node based on its {@link StoredKey | stored key}, if it exists.
+	 *
+	 * @param node - The node whose child is being queried.
+	 * @param storedKey - The stored key whose associated child will be returned.
+	 *
+	 * @returns The associated child node, if it exists. Otherwise `undefined`.
+	 */
+	child(node: TreeNode, storedKey: StoredKey): TreeNode | TreeValue | undefined;
+
 	/**
 	 * Return the node under which this node resides in the tree (or undefined if this is a root node of the tree).
 	 */
 	parent(node: TreeNode): TreeNode | undefined;
+
 	/**
 	 * The key of the given node under its parent.
 	 * @remarks
 	 * If `node` is an element in a {@link (TreeArrayNode:interface)}, this returns the index of `node` in the array node (a `number`).
 	 * Otherwise, this returns the key of the field that it is under (a `string`).
 	 */
 	key(node: TreeNode): string | number;
+
+	/**
+	 * Gets the {@link StoredKey | stored key} of the given node under its parent.
+	 *
+	 * @remarks
+	 * If one was not explicitly specified by the schema of the parent node, this will be the same as the
+	 * object property key.
+	 *
+	 * @param node - The node whose stored key is being queried.
+	 */
+	storedKey(node: TreeNode): StoredKey | number;
```

Comment:
Would it make sense to document that it will always return the `rootFieldKey` constant? We have test coverage for this. We don't currently (publicly) export that constant, but maybe it would make sense to do so?

Reply:
For the root case, maybe we want to stabilize that. Note that constant is currently internal.

For roots of removed content, they would have to return something else.

What ever we do, ideally the inverse of this Tree.child(Tree.parent(x), tree.stableKey(x)) would get us our x back. With the current API, that would require returning an object with a reference to the view since parent returns undefined. This sounds messy. This is one of the cases where not having field objects is a real pain.

Reply:
We've opted to remove these new functions for now.